router/network.go

```diff
@@ -157,6 +187,9 @@ func (network *netHandle) SendPost(ctx context.Context, structData integrations.
 		}
 
 		req.Header.Add("User-Agent", "RudderLabs")
+		if bodyFormat == "GZIP" {
+			req.Header.Add("Content-Encoding", "gzip")
+		}
```

Comment:
instead of adding another if clause here, we could use a headers map and fill it per case, e.g.

```go
var payload io.Reader
headers := map[string]string{"User-Agent": "RudderLabs"}
...
switch bodyFormat {
	case "GZIP":
		headers["Content-Encoding"] = "gzip"
...
for key, val := range headers {
	req.Header.Add(key, val)
}
...
```

Reply:
incorporated the review suggestions